Sources/PortalSwift/Portal.swift

```diff
@@ -30,32 +30,76 @@ public final class Portal: PortalProtocol {
     self.provider.chainId
   }
 
-  public let api: PortalApiProtocol
+  public private(set) var api: PortalApiProtocol
   public let apiKey: String
   public let autoApprove: Bool
   public var gatewayConfig: [Int: String] = [:]
   public var provider: PortalProviderProtocol
   public var rpcConfig: [String: String]
 
-  private let apiHost: String
+  private var apiHost: String
   private var backup: BackupOptions?
-  private let binary: Mobile
+  private var binary: Mobile
   private let featureFlags: FeatureFlags?
   private var keychain: PortalKeychainProtocol
-  private let mpc: PortalMpcProtocol
-  private let mpcHost: String
-  private let version: String
+  private var mpc: PortalMpcProtocol
+  private var mpcHost: String
+  private var version: String
+  private var enclaveMPCHost: String
+
+  /// Creates a Portal instance with the essential configuration.
+  /// This is the recommended initializer for most use cases.
+  /// - Parameters:
+  ///   - apiKey: The Client API key. You can obtain this through Portal's REST API.
+  ///   - withRpcConfig: (optional) A dictionary of CAIP-2 Blockchain IDs (keys) and RPC URLs (values) in `[String:String]` format.
+  ///   - autoApprove: (optional) When true, transactions will be automatically approved without user confirmation.
+  ///   - featureFlags: (optional) A set of flags to opt into new or experimental features.
+  ///   - version: (optional) The API version to use. Currently only "v6" is supported.
+  ///   - apiHost: (optional) Portal's API host. Defaults to "api.portalhq.io".
+  ///   - mpcHost: (optional) Portal's MPC API host. Defaults to "mpc.portalhq.io".
+  /// - Throws: `PortalArgumentError` if an unsupported version is provided.
+  public convenience init(
```

Comment:
lets add this init to the example app somehow. Possibly via a switch we can tell the example app to use this